com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -2973,6 +2986,12 @@ public void Serialize(FastBufferWriter writer)
                     writer.WriteValue(OwnerObject.GetSceneOriginHandle());
                 }
 
+                if (HasInstantiationData)
+                {
+                    writer.WriteValueSafe(OwnerObject.InstantiationData.Length);
```

Comment:
The Length will also be written inside `writer.WriteBytesSafe()` as a `uint` (It's under a few layers of function calls, but ultimately a byte array will be serialized [here](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/blob/e3189a26d35bb4087aa092f856ee55e5f4fb0ac6/com.unity.netcode.gameobjects/Runtime/Serialization/FastBufferWriter.cs#L963-L972)). I'm not sure if it's over complicating things to use that length on the deserialize?

Reply:
I’d rather not switch to BytePacker utilities here as `ByteUnpacker` doesn’t support reading a `T : INetworkSerializable` directly. I’d have to manually extract the `byte[]` during the instantiation call, then marshalling, which would bypass the `INetworkSerializable.NetworkSerialize()` logic, effectively breaking any user-defined serialization code inside that method.